package/MDAnalysis/lib/pkdtree.py

```diff
@@ -0,0 +1,221 @@
+        # extents marks the max distance to plane just to consider an image
+        extents = np.array([norm(self._dm[i])/2.0 for i in range(self.dim)])
+        extents = np.where(extents < radius, extents, radius)
+        # displacements are vectors that we add to wrapped_c to
+        # generate images "up" or "down" the central cell along
+        # the axis that we happen to be looking.
+        displacements = list()
+        for i in range(self.dim):
+            # distance to the plane containing the origin
+            if np.dot(wrapped_c, self._rm[i]) < extents[i]:
+                displacements.append(self._dm[i])  # "upper" image
+            # distance to the plane containing point self._dm[i]
+            elif np.dot(self._dm[i] - wrapped_c, self._rm[i]) < extents[i]:
+                displacements.append(-self._dm[i])  # "lower" image
+        # If we have displacements along more than one axis, we have
+        # to combine them. This happens when wrapped_center is close
+        # to any edge or vertex of the central cell.
+        # face, n_displacements==1; no combination
+        # edge, n_displacements==2; combinations produce one extra displacement
+        # vertex, n_displacements==3; five extra displacements
+        n_displacements = len(displacements)
+        if n_displacements > 1:
+            for start in range(n_displacements - 1, -1, -1):
+                for i in range(start+1, len(displacements)):
+                    displacements.append(displacements[start]+displacements[i])
+        return [wrapped_c, ] + [wrapped_c + d for d in displacements]
+
+    def search(self, center, radius):
+        """Search all points within radius of center and its periodic images.
+        Wrapping of center coordinates is enforced to enable comparison to
+        wrapped coordinates of points in the tree.
+
+        Parameter
+        ---------
+        center: NumPy.array
+          origin around which to search for neighbors
+        radius: float
+          maximum distance around which to search for neighbors. The search
+          radius is half the smallest periodicity if radius exceeds this value
+        """
+        if not self.built:
+            raise RuntimeError('Unbuilt tree. Run tree.set_coords first')
+        if center.shape != (self.dim,):
+            raise ValueError('Expected a ({},) NumPy array'.format(self.dim))
+        self._indices = set()  # clear previous search
+        for c in self.find_centers(center, radius):
+            self.kdt.search_center_radius(c, radius)
+            new_indices = self.kdt.get_indices()  # returns None or np.array
+            if new_indices is not None:
+                self._indices.update(new_indices)
+        if self._indices:
+            self._indices = sorted(list(self._indices))
+        else:
+            # Bio.KDTree.KDTree.getIndices returns None if no neighbors are found
+            self._indices = None
```

Comment:
This should be an empty list like the Bio.KDTree returns. Please also add a test that `_indices` is a list even if we don't find any neighbors.

Reply:
That _indices is a list object is an assumption we rely on when we use the KDTree algorithms.

Reply:
@kain88-de you're correct, I wrongly thought it returned None! Well, this simplifies things :smile: